astropy/coordinates/transformations.py

```diff
@@ -1044,99 +1009,90 @@ def __call__(self, fromcoord, toframe):
         from .representation import CartesianDifferential, CartesianRepresentation
 
         supcall = self.func
-        if fromcoord.data.differentials:
-            # this is the finite difference case
-
-            if callable(self.finite_difference_dt):
-                dt = self.finite_difference_dt(fromcoord, toframe)
-            else:
-                dt = self.finite_difference_dt
-            halfdt = dt / 2
+        if not fromcoord.data.differentials:
+            return supcall(fromcoord, toframe)
+        # this is the finite difference case
 
-            from_diffless = fromcoord.realize_frame(
-                fromcoord.data.without_differentials()
+        if callable(self.finite_difference_dt):
+            dt = self.finite_difference_dt(fromcoord, toframe)
```

Comment:
I don't like the lack of test coverage for this line.

Reply:
Although it would be very simple to modify a test so that this line would be executed in CI, I think it would be better to write a non-trivial test that would actually check the underlying code. I also think that such a change would be large enough to deserve its own pull request. @mhvk, do you agree with merging this pull request leaving the test coverage as it is for now?

Reply:
Yes, I think that is fine. I certainly agree that a trivial test has little purpose -- the code is fairly obviously correct.